### DISCUSSION TITLE
eq method on tuples

### DISCUSSION THREAD

**Discussion by dullbananas** [diesel-rs/diesel#3765] [Ideas]
`(a1, b1).eq((a2, b2))` would be much cleaner than `a1.eq(a2).and(b1.eq(b2))`.

More importantly, this would also allow using the type `dsl::Eq<(a1, b1), (a2, b2)>` instead of `dsl::And<dsl::Eq<a1, a2>, dsl::Eq<b1, b2>>`.

**Comment by weiznich** [comment]
Duplicate of https://github.com/diesel-rs/diesel/discussions/3723
